From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Emilio=20Cobos=20=C3=81lvarez?= <emilio@crisal.io>
Date: Wed, 15 Jul 2026 17:47:37 +0200
Subject: [PATCH] Bug 2055018 - Add some missing X11UndefineNone includes.
 r=stransky

And clean up some unused ones while at it.

Differential Revision: https://phabricator.services.mozilla.com/D312311
---
 widget/gtk/WidgetUtilsGtk.cpp       | 13 +++++------
 widget/gtk/nsPrintDialogGTK.cpp     | 35 ++++++++++-------------------
 widget/gtk/nsUserIdleServiceGTK.cpp | 21 ++++++++---------
 3 files changed, 27 insertions(+), 42 deletions(-)

diff --git a/widget/gtk/WidgetUtilsGtk.cpp b/widget/gtk/WidgetUtilsGtk.cpp
index 516b259547c1..f5bab748a847 100644
--- a/widget/gtk/WidgetUtilsGtk.cpp
+++ b/widget/gtk/WidgetUtilsGtk.cpp
@@ -8,40 +8,39 @@
 #include <glib.h>
 #include <gtk/gtk.h>
 #include <inttypes.h>
+#ifdef MOZ_X11
+#  include <X11/Xatom.h>
+#  include <X11/Xlib.h>
+
+#  include "X11UndefineNone.h"
+#endif
 
 #include "MainThreadUtils.h"
-#include "mozilla/Components.h"
 #include "mozilla/StaticPrefs_widget.h"
 #include "mozilla/UniquePtr.h"
 #include "nsCOMPtr.h"
 #include "nsDirectoryServiceDefs.h"
 #include "nsGtkKeyUtils.h"
 #include "nsGtkUtils.h"
 #include "nsIFile.h"
-#include "nsIGfxInfo.h"
 #include "nsIProperties.h"
 #include "nsReadableUtils.h"
 #include "nsString.h"
 #include "nsStringFwd.h"
 #include "nsWindow.h"
 #include "nsXPCOMCID.h"
 #include "nsXULAppAPI.h"
 
 #ifdef MOZ_ENABLE_DBUS
 #  include "mozilla/ClearOnShutdown.h"
 #  include "mozilla/widget/AsyncDBus.h"
 #  include "nsAppShell.h"
 #endif  // MOZ_ENABLE_DBUS
 
 #ifdef MOZ_WAYLAND
 #  include "nsWaylandDisplay.h"
 #endif  // MOZ_WAYLAND
 
-#ifdef MOZ_X11
-#  include <X11/Xatom.h>
-#  include <X11/Xlib.h>
-#endif /* MOZ_X11 */
-
 #undef LOGW
 #ifdef MOZ_LOGGING
 #  include "mozilla/Logging.h"
diff --git a/widget/gtk/nsPrintDialogGTK.cpp b/widget/gtk/nsPrintDialogGTK.cpp
index 6e2d9280a099..c3084f02e0dd 100644
--- a/widget/gtk/nsPrintDialogGTK.cpp
+++ b/widget/gtk/nsPrintDialogGTK.cpp
@@ -4,47 +4,36 @@
 
 #include "nsPrintDialogGTK.h"
 
+#include <gdk/gdk.h>  // for gdk_x11_window_get_xid
 #include <gtk/gtk.h>
 #include <gtk/gtkunixprint.h>
 #include <stdlib.h>
+#ifdef MOZ_X11
+#  include <gdk/gdkx.h>
+
+#  include "X11UndefineNone.h"
+#endif
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <gio/gunixfdlist.h>
+#include <sys/stat.h>
+#include <sys/types.h>
 
 #include "GRefPtr.h"
-#include "MozContainer.h"
+#include "MainThreadUtils.h"
 #include "WidgetUtils.h"
-#include "WidgetUtilsGtk.h"
-#include "mozilla/Services.h"
 #include "mozilla/dom/Promise.h"
-#include "nsIGIOService.h"
 #include "nsIGlobalObject.h"
-#include "nsIObserverService.h"
 #include "nsIPrintSettings.h"
 #include "nsIPrintSettingsService.h"
 #include "nsIStringBundle.h"
 #include "nsIWidget.h"
 #include "nsPIDOMWindow.h"
 #include "nsPrintSettingsGTK.h"
-#include "nsPrintfCString.h"
-#include "nsReadableUtils.h"
 #include "nsServiceManagerUtils.h"
 #include "nsString.h"
-#include "nsThreadUtils.h"
 #include "xpcpublic.h"
 
-// for gdk_x11_window_get_xid
-#include <gdk/gdk.h>
-#ifdef MOZ_X11
-#  include <gdk/gdkx.h>
-#endif
-#include <fcntl.h>
-#include <gio/gunixfdlist.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-
-// for dlsym
-#include <dlfcn.h>
-
-#include "MainThreadUtils.h"
-
 using namespace mozilla;
 using namespace mozilla::dom;
 using namespace mozilla::widget;
diff --git a/widget/gtk/nsUserIdleServiceGTK.cpp b/widget/gtk/nsUserIdleServiceGTK.cpp
index 63a55fa8c768..334be476dfb6 100644
--- a/widget/gtk/nsUserIdleServiceGTK.cpp
+++ b/widget/gtk/nsUserIdleServiceGTK.cpp
@@ -5,27 +5,24 @@
 #include "nsUserIdleServiceGTK.h"
 
 #include <gtk/gtk.h>
-
-#include "WidgetUtilsGtk.h"
-#include "mozilla/Logging.h"
-#include "mozilla/SpinEventLoopUntil.h"
-#include "nsDebug.h"
-#include "nsITimer.h"
-#include "prlink.h"
 #ifdef MOZ_X11
 #  include <X11/Xlib.h>
 #  include <X11/Xutil.h>
 #  include <gdk/gdkx.h>
+
+#  include "X11UndefineNone.h"
 #endif
+
+#include "WidgetUtilsGtk.h"
+#include "mozilla/Logging.h"
+#include "mozilla/SpinEventLoopUntil.h"
+#include "nsITimer.h"
+#include "prlink.h"
+
 #ifdef MOZ_ENABLE_DBUS
-#  include <gio/gio.h>
-
 #  include "AsyncDBus.h"
-#  include "WakeLockListener.h"
-#  include "nsIObserverService.h"
 #endif
 
-using mozilla::LogLevel;
 static mozilla::LazyLogModule sIdleLog("nsIUserIdleService");
 
 using namespace mozilla;
